retryable_http_client.go

```diff
@@ -0,0 +1,58 @@
+package statuspage
+
+import (
+	"context"
+	"github.com/hashicorp/go-retryablehttp"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+func NewRetryableClient() *retryablehttp.Client {
+	client := retryablehttp.NewClient()
+	client.CheckRetry = retryPolicy
+	client.Backoff = backoffPolicy
+	return client
+}
+
+const (
+	RateLimitExceededStatusCode = 420
+)
+
+// We're wrapping original retryablehttp.DefaultRetryPolicy and adding retry on 420 HTTP code
+func retryPolicy(ctx context.Context, resp *http.Response, err error) (bool, error) {
+	defaultShouldRetry, _ := retryablehttp.DefaultRetryPolicy(ctx, resp, err)
+	isRateLimited := resp != nil && resp.StatusCode == RateLimitExceededStatusCode
+	shouldRetry := defaultShouldRetry || isRateLimited
+	return shouldRetry, nil
+}
+
+// We're wrapping original retryablehttp.DefaultBackoff and using response header in such format `Retry-After: 60`
```

Comment:
Not that keen on golang, but can't we just override `retryablehttp` and override `DefaultBackoff`?